### DISCUSSION TITLE
Variable naming

### DISCUSSION THREAD

**Discussion by fdncred** [nushell/nushell#6459] [Polls]
A recent PR was landed that changes variables to be referenced without a `$` prefix.

Example 1:
```
# with dollar sign prefix
let x = 100
print $x
#
```
Example 2:
```
# without dollar sign prefix
let x = 100
print x
```

**Comment by belchior** [comment]
If possible keep both to interrop with old shells

**Comment by adamijak** [comment]
```
# with dollar sign prefix everywhere
let $x = 100
print $x
#
```

**Comment by rgwood** [comment]
Unsure which is best. I started off not liking the `$`, but now I find it useful to disambiguate variables from other things (bare words, command names).

**Reply by rgwood** [reply]
Just watching JT's latest video now; I agree that IDE support/highlighting greatly improves life without the `$` https://www.youtube.com/watch?v=sSopO8x755I

**Reply by adamijak** [reply]
Are there any terminals which do not support colors? Are we sure syntax highlighting will be available everywhere?

**Reply by fdncred** [reply]
yes, there are terminals that do not support color which the reason why we have a `use_ansi_coloring` config flag, so those people can turn it off.

**Comment by hustcer** [comment]
It's a good idea to use `$ENV_VAR` instead of `$env.ENV_VAR`

**Reply by kubouch** [reply]
Yes, if we end up without requiring the `$`, the `$` could be used to denote environment variables.

**Reply by adamijak** [reply]
Should not we use `$` for each variable or not use it at all? For sake of consistency.

**Comment by Zoybean** [comment]
My initial preference was for without `$`, but only so long as it is able to be unambiguous with command / subcommand names, bare words, etc. Not just to the compiler, but ideally to readers too (not requiring syntax highlighting). But the more I think about it, the more I think I value the ease of passing strings and commands around unquoted, over the ease of referring to variables without $.

**Comment by generic-github-user** [comment]
I'd prefer having both available and leaving it as a matter of style where unambiguous.
